Smartway.UiComponent/Inputs/NumericDateEntry.xaml.cs

```diff
@@ -0,0 +1,120 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Windows.Input;
+using Xamarin.Forms;
+using Xamarin.Forms.Xaml;
+
+namespace Smartway.UiComponent.Inputs
+{
+    [XamlCompilation(XamlCompilationOptions.Compile)]
+    public partial class NumericDateEntry
+    {
+        public static readonly BindableProperty GlobalFontSizeProperty =
+            BindableProperty.Create(nameof(GlobalFontSize), typeof(int), typeof(NumericDateEntry), 13);
+
+        public static readonly BindableProperty FilledDateTimeProperty =
+            BindableProperty.Create(nameof(FilledDateTime), typeof(DateTime), typeof(NumericDateEntry), defaultBindingMode: BindingMode.TwoWay);
+
+        public static readonly BindableProperty ErrorCommandProperty =
+            BindableProperty.Create(nameof(ErrorCommand), typeof(ICommand), typeof(NumericDateEntry), null);
+
+
+        public int GlobalFontSize
+        {
+            get => (int)GetValue(GlobalFontSizeProperty);
+            set => SetValue(GlobalFontSizeProperty, value);
+        }
+
+        public DateTime FilledDateTime 
+        {
+            get => (DateTime)GetValue(FilledDateTimeProperty);
+            set => SetValue(FilledDateTimeProperty, value);
+        }
+
+        public ICommand ErrorCommand
+        {
+            get => (ICommand)GetValue(ErrorCommandProperty);
+            set => SetValue(ErrorCommandProperty, value);
+        }
+
+        private List<Entry> _dateEntries;
+        public NumericDateEntry()
+        {
+            InitializeComponent();
+
+            _dateEntries = new List<Entry> {
+                FirstEntry,
+                SecondEntry,
+                ThirdEntry
+            };
+            SetFilledDate();
+            SetEventsOnDateEntries();
+        }
+
+        private void SetFilledDate()
+        {
+            //Find how to treat nullable datetime
+            if (FilledDateTime.Year == 1 )
+                return;
+
+            var filledDateTime = FilledDateTime;
+            FirstEntry.Text = filledDateTime.Day.ToString();
+            SecondEntry.Text = filledDateTime.Month.ToString();
+            ThirdEntry.Text = filledDateTime.Year.ToString().Substring(2);
+        }
+
+        private void SetEventsOnDateEntries()
+        {
+            _dateEntries.ForEach(_ => _.PropertyChanged += EntryOnPropertyChanged);
+        }
+
+        private void EntryOnPropertyChanged(object sender, EventArgs e)
+        {
+            var currentEntry = sender as Entry;
+            if (IsValidDateEntry(currentEntry))
+                return;
+
+            var hasFocusEmptyEntry = FocusNextEmptyEntry();
```

Comment:
why do we need a variable alors qu'on l'utiliser pas ailleurs apart le if?

Reply:
C'est pour clarifier le code car ce n'est pas forcément évident de dire que en nom de fonction qu'il retourne un bool. Il y a peut être une meilleur solution ?

Reply:
on pourrait pas mêttre a private property ? (je trouve ça plus beau)

`private bool hasFocusEmptyEntry => FocusNextEmptyEntry(); `

Reply:
bof c'est utilisé que dans cette méthode, en plus va falloir gérer le reset pas trop fan :/